cmd/utilities/utilities.go

```diff
@@ -0,0 +1,200 @@
+package utilities
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/zebedee"
+	"github.com/ONSdigital/dp-net/v2/awsauth"
+	dphttp "github.com/ONSdigital/dp-net/v2/http"
+	"github.com/ONSdigital/dp-sitemap/assets"
+	"github.com/ONSdigital/dp-sitemap/config"
+	"github.com/ONSdigital/dp-sitemap/event"
+	"github.com/ONSdigital/dp-sitemap/robotseo"
+	"github.com/ONSdigital/dp-sitemap/sitemap"
+	es710 "github.com/elastic/go-elasticsearch/v7"
+)
+
+// Config represents service configuration for dp-sitemap
+type FlagFields struct {
+	RobotsFilePath string // path to the robots file
+	APIURL         string // elastic search api url
+	SitemapIndex   string // elastic search sitemap index
+	ScrollTimeout  string // elastic search scroll timeout
+	ScrollSize     int    // elastic search scroll size
+	SitemapPath    string // path to the sitemap file
+	ZebedeeURL     string // zebedee url
+	FakeScroll     bool   // toggle to use or not the fake scroll implementation that replicates elastic search
+}
+
+func GenerateSitemap(cfg *config.Config, commandline *FlagFields) {
+	store := &sitemap.LocalStore{}
+
+	var transport http.RoundTripper = dphttp.DefaultTransport
+
+	// add SignerRegion,SignerService
+	if cfg.OpenSearchConfig.Signer {
+		var err error
+		transport, err = awsauth.NewAWSSignerRoundTripper(cfg.OpenSearchConfig.APIURL, cfg.OpenSearchConfig.SignerFilename, cfg.OpenSearchConfig.SignerRegion, cfg.OpenSearchConfig.SignerService, awsauth.Options{TlsInsecureSkipVerify: cfg.OpenSearchConfig.TLSInsecureSkipVerify})
+		if err != nil {
+			fmt.Printf("failed to save file")
+			return
+		}
+	}
+
+	rawClient, err := es710.NewClient(es710.Config{
+		Addresses: []string{commandline.APIURL},
+		Transport: transport,
+	})
+	if err != nil {
+		return
+	}
+
+	// Get zebedeeClient using arg -zebedee-url
+	zebedeeClient := zebedee.New(commandline.ZebedeeURL)
+
+	var scroll sitemap.Scroll
+	if commandline.FakeScroll {
+		scroll = NewFakeScroll()
+	} else {
+		scroll = sitemap.NewElasticScroll(
+			rawClient,
+			cfg,
+		)
+	}
+
+	generator := sitemap.NewGenerator(
+		sitemap.WithFetcher(sitemap.NewElasticFetcher(
+			scroll,
+			cfg,
+			zebedeeClient,
+		)),
+		sitemap.WithAdder(&sitemap.DefaultAdder{}),
+		sitemap.WithFileStore(store),
+		sitemap.WithFullSitemapFiles(map[config.Language]string{
+			config.English: commandline.SitemapPath + "_en",
+			config.Welsh:   commandline.SitemapPath + "_cy",
+		}),
+	)
+
+	// Generating sitemap
+	genErr := generator.MakeFullSitemap(context.Background())
+	if genErr != nil {
+		fmt.Println("Error writing sitemap file", genErr.Error())
+		return
+	}
+
+	GenerateRobotFile(cfg, commandline)
```

Comment:
Would be sensible to separate this out and not have the sitemap creation call the robots creation I think?